galsim/sed.py

```diff
@@ -292,47 +489,58 @@ def __add__(self, other):
         # 1) The SEDs must have the same redshift.
         # 2) The resulting SED will be defined on the wavelength range set by the overlap of the
         #    wavelength ranges of the two SED operands.
-        # 3) If both SEDs maintain a `wave_list` attribute, then the new `wave_list` will be
-        #    the union of the old `wave_list`s in the intersecting region.
-        # This ensures that SED addition is commutative.
+        # 3) The new `wave_list` will be the union of the operand `wave_list`s in the intersecting
+        #    region, even if one or both of the `wave_list`s are empty.
+        # These conditions ensure that SED addition is commutative.
 
         if self.redshift != other.redshift:
             raise ValueError("Can only add SEDs with same redshift.")
 
-        blue_limit, red_limit = self._wavelength_intersection(other)
-        spec = lambda w: self._rest_photons(w) + other._rest_photons(w)
-        wave_list = np.union1d(self.wave_list, other.wave_list)
-        wave_list = wave_list[wave_list <= red_limit]
-        wave_list = wave_list[wave_list >= blue_limit]
+        if self.dimensionless and other.dimensionless:
+            flux_type = '1'
+            _spectral = False
+            _dimensionless = True
+        elif self.spectral and other.spectral:
+            flux_type = 'fphotons'
+            _spectral = True
+            _dimensionless = False
+        else:
+            raise TypeError("Cannot add SEDs with incompatible dimensions.")
 
-        return SED(spec, wave_type='nm', flux_type='fphotons',
+        spec = lambda w: self(w*(1.0+self.redshift)) + other(w*(1.0+self.redshift))
+        wave_list, blue_limit, red_limit = galsim.utilities.combine_wave_list([self, other])
+
+        return SED(spec, wave_type='nm', flux_type=flux_type,
                    redshift=self.redshift, _wave_list=wave_list,
-                   _blue_limit=blue_limit, _red_limit=red_limit)
+                   _blue_limit=blue_limit, _red_limit=red_limit,
+                   _spectral=_spectral, _dimensionless=_dimensionless)
 
     def __sub__(self, other):
         # Subtract two SEDs, with the same caveats as adding two SEDs.
         return self.__add__(-1.0 * other)
 
-    def copy(self):
-        import copy
-        return copy.deepcopy(self)
```

Comment:
I'd like to wait to drop 1.x deprecations until 2.0.  Ideally, any code that worked with a 1.x version of GalSim will keep working (although I doubt that is strictly true...).  Then at version 2.0, people have to adapt to all the new conventions.

But I'm fine with moving the copy() functions to deprecated and just point out that they are unnecessary unless you are doing something very weird with the internals of an object.